### PR TITLE
client: install metainfo file

### PR DESCRIPTION
**Pull request by allisonkarlitskaya**
Create a tool to turn the point form notes in our tags into a proper set
of `<release/>` tags for metainfo and fill this information in when
building a tarball.

Install the metainfo if building with --enable-client.

This will simplify the process of releasing the flatpak.

### CODE REVIEW

File: containers/flatpak/prepare [hunk 1]
```diff
@@ -78,8 +58,14 @@ prepare() {
 
     TODAY="$(date +%F)"
 
-    subst metainfo.xml
-    subst yml
+    sed \
```
Comment (martinpitt) [inline containers/flatpak/prepare:61]: I kinda liked the separate `subst()` function -- shortens the function a bit, and we might need it for other files, too. But bikeshedding, and very weak opinion.

File: src/client/add-metainfo-releases [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    tree = ET.parse(args.xml)
+    releases = tree.find('releases')
+
+    # Only update the file if it doesn't already contain releases.
+    if not releases.find('release'):
+        if 'g' in args.version:
+            today = time.strftime('%Y-%m-%d')
+            releases.append(element('release', version=args.version, date=today, type='development'))
+        releases.extend(format_release(r) for r in get_releases())
+        ET.indent(releases, space="  ", level=1)
```
Comment (martinpitt) [inline src/client/add-metainfo-releases:55]: This is not present in Ubuntu 20.04 yet, see [failed workflow](https://github.com/cockpit-project/cockpit/runs/5773904226?check_suite_focus=true). As 20.04 LTS is a very popular platform for all kinds of infrastructure, let's find an API that works there. I'm sure ElementTree has *some* pretty-printer?

File: src/client/add-metainfo-releases [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    url = element('url', f'https://cockpit-project.org/blog/cockpit-{version}.html', type='details')
+    ul = element('ul', children=points_from_body(body))
+    description = element('description', children=(ul,))
+    return element('release', version=version, date=date, children=(url, description))
+
+
+def get_releases():
+    format = '%(if)%(taggerdate)%(then)%(refname:short)|%(taggerdate:short)|%(contents:body)%(end)'
+    output = subprocess.check_output(['git', 'for-each-ref', '--shell', '--sort=-taggerdate',
+                                      '--contains', '258', '--format', format, 'refs/tags'], text=True)
```
Comment (martinpitt) [inline src/client/add-metainfo-releases:36]: What's the magic "258" here? Shouldn't that be the `--version` value?

**Review by martinpitt — CHANGES_REQUESTED** [review]
Nice!

File: src/client/add-metainfo-releases (hunk 3 above)
Comment (allisonkarlitskaya) [reply src/client/add-metainfo-releases:37]: This is when we started shipping the flatpak.

Comment (allisonkarlitskaya) [reply src/client/add-metainfo-releases:39]: Added a comment about it.

Comment (martinpitt) [reply src/client/add-metainfo-releases:42]: Oh, I see -- so for each release, we have to regenerate the entire release history. I forgot that, but makes sense now.

File: .github/workflows/flatpak-test.yml [hunk 4]
```diff
@@ -24,10 +24,7 @@ jobs:
           fetch-depth: 0
 
       - name: Build and install flatpak
-        run: sh -x containers/flatpak/install --user --install-deps-from=flathub
-
-      - name: Validate metainfo
-        run: appstream-util validate org.cockpit_project.CockpitClient.metainfo.xml
+        run: ELEMENT_TREE_NO_INDENT=1 sh -x containers/flatpak/install --user --install-deps-from=flathub
```
Comment (martinpitt) [inline .github/workflows/flatpak-test.yml:27]: That could have used a comment, but we can always get it from git blame.

**Review by martinpitt — APPROVED** [review]
Cheers!

**Pull request merged**